test/hotspot/jtreg/serviceability/attach/AttachAPIv2/CompatTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Basic compatibility test for Attach API v2
+ * @bug 8219896
+ * @library /test/lib
+ * @modules jdk.attach/sun.tools.attach
+ *
+ * @run main/othervm -Xlog:attach=trace CompatTest
+ * @run main/othervm -Xlog:attach=trace -Djdk.attach.compat=true CompatTest
+ */
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.tools.attach.HotSpotVirtualMachine;
+
+import jdk.test.lib.apps.LingeredApp;
+
+public class CompatTest {
+
+    public static void main(String[] args) throws Exception {
+        // if the test (client part) in the "compat" mode
+        boolean clientCompat = "true".equals(System.getProperty("jdk.attach.compat"));
```

Comment:
Boolean.getBoolean("jdk.attach.compat") ignores case for us as well.
But more generally, are we happy with "jdk.attach.compat" as a boolean flag, or would "jdk.attach.version" be better?
(which is mainly asking if we think there might ever be another version!)

Reply:
I thought about making the property numeric, but that adds complexity in the logic and I don't think  we will need another version soon. Anyway it's internal property for testing only, so we can change it any time.